whois-client/src/main/java/net/ripe/db/whois/common/sso/AuthServiceClient.java

```diff
@@ -0,0 +1,195 @@
+            throw new AuthServiceClientException(UNAUTHORIZED.getStatusCode(), "Invalid username.");
+        } catch (WebApplicationException e) {
+            LOGGER.debug("Failed to get details for email {} due to {}:{}\n\tResponse: {}", username, e.getClass().getName(), e.getMessage(), e.getResponse().readEntity(String.class));
+            throw new AuthServiceClientException(INTERNAL_SERVER_ERROR.getStatusCode(), "Internal server error");
+        } catch (ProcessingException e) {
+            LOGGER.debug("Failed to get details for email {} due to {}:{}", username, e.getClass().getName(), e.getMessage());
+            throw new AuthServiceClientException(INTERNAL_SERVER_ERROR.getStatusCode(), "Internal server error");
+        }
+    }
+
+    public ValidateTokenResponse getUserDetails(final String uuid) {
+        if (StringUtils.isEmpty(uuid)) {
+            LOGGER.info("No uuid was supplied");
+            throw new NotAuthorizedException("Invalid uuid.");
+        }
+
+        try {
+            return client.target(restUrl)
+                    .path(USER_SEARCH_PATH)
+                    .path(uuid)
+                    .request(MediaType.APPLICATION_JSON_TYPE)
+                    .header("X-API_KEY", apiKey)
+                    .get(ValidateTokenResponse.class);
+        } catch (BadRequestException e) {
+            LOGGER.debug("Failed to get details for uuid {} (token is invalid)", uuid);
+            throw new NotAuthorizedException("Invalid token.");
+        } catch (WebApplicationException e) {
+            LOGGER.debug("Failed to get details for uuid {} due to {}:{}\n\tResponse: {}", uuid, e.getClass().getName(), e.getMessage(), e.getResponse().readEntity(String.class));
+            throw new AuthServiceClientException(INTERNAL_SERVER_ERROR.getStatusCode(), "Internal server error");
+        } catch (ProcessingException e) {
+            LOGGER.debug("Failed to get details for uuid {} due to {}:{}", uuid, e.getClass().getName(), e.getMessage());
+            throw new AuthServiceClientException(INTERNAL_SERVER_ERROR.getStatusCode(), "Internal server error");
+        }
+    }
+
+    public String getUsername(final String uuid) {
+        return getUserDetails(uuid).response.content.email;
+    }
+
+    public String getDisplayName(final String uuid) {
+        return getUserDetails(uuid).response.content.getName();
+    }
+
+    public List<String> getOrgsContactsEmails(final Long membershipId) {
+        if (membershipId != null) {
+            try {
+                final MemberContactsResponse response = client.target(restUrl)
+                        .path(ORGANISATION_MEMBERS_PATH)
+                        .path(String.valueOf(membershipId))
+                        .path(CONTACT_PATH)
+                        .queryParam("api-key", apiKey) // TODO remove this when this endpoint supports apikey as a header
```

Comment:
Check if we can replace "api-key" with an API key header

Reply:
Confirmed (tested) that the "api-key" query parameter can be replaced with the X-API_KEY header

Reply:
We are using header X-API_KEY only